diff -Nuar v4l-utils-0.8.4-orig/utils/decode_tm6000/Makefile v4l-utils-0.8.4/utils/decode_tm6000/Makefile
--- v4l-utils-0.8.4-orig/utils/decode_tm6000/Makefile	2011-06-01 12:23:03.000000000 +0100
+++ v4l-utils-0.8.4/utils/decode_tm6000/Makefile	2011-06-29 23:23:17.721058396 +0100
@@ -5,7 +5,7 @@
 -include *.d
 
 decode_tm6000: decode_tm6000.o ../libv4l2util/libv4l2util.a
-	$(CC) $(LDFLAGS) -o $@ $^
+	$(CC) $(LDFLAGS) -o $@ $^ -largp
 
 install: $(TARGETS)
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
diff -Nuar v4l-utils-0.8.4-orig/utils/keytable/Makefile v4l-utils-0.8.4/utils/keytable/Makefile
--- v4l-utils-0.8.4-orig/utils/keytable/Makefile	2011-06-01 12:23:03.000000000 +0100
+++ v4l-utils-0.8.4/utils/keytable/Makefile	2011-06-29 23:23:43.041058396 +0100
@@ -5,7 +5,7 @@
 -include *.d
 
 ir-keytable: keytable.o
-	$(CC) $(LDFLAGS) -o $@ $^
+	$(CC) $(LDFLAGS) -o $@ $^ -largp
 
 keytable.o: keytable.c parse.h
 
diff -Nuar v4l-utils-0.8.4-orig/utils/v4l2-compliance/Makefile v4l-utils-0.8.4/utils/v4l2-compliance/Makefile
--- v4l-utils-0.8.4-orig/utils/v4l2-compliance/Makefile	2011-06-01 12:23:03.000000000 +0100
+++ v4l-utils-0.8.4/utils/v4l2-compliance/Makefile	2011-06-29 23:42:07.217058397 +0100
@@ -6,7 +6,7 @@
 
 v4l2-compliance: v4l2-compliance.o v4l2-test-debug.o v4l2-test-input-output.o \
 	v4l2-test-controls.o v4l2-test-io-config.o
-	$(CXX) $(LDFLAGS) -o $@ $^ -lv4l2 -lv4lconvert -lrt
+	$(CXX) $(LDFLAGS) -o $@ $^ -lv4l2 -lv4lconvert -lrt -ljpeg
 
 install: $(TARGETS)
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
diff -Nuar v4l-utils-0.8.4-orig/utils/v4l2-ctl/Makefile v4l-utils-0.8.4/utils/v4l2-ctl/Makefile
--- v4l-utils-0.8.4-orig/utils/v4l2-ctl/Makefile	2011-06-01 12:23:03.000000000 +0100
+++ v4l-utils-0.8.4/utils/v4l2-ctl/Makefile	2011-06-29 23:56:12.557058397 +0100
@@ -13,7 +13,7 @@
 	$(CC) $(LDFLAGS) -o $@ $^ -lm
 
 v4l2-ctl: v4l2-ctl.o
-	$(CXX) $(LDFLAGS) -o $@ $^ -lv4l2 -lv4lconvert -lrt
+	$(CXX) $(LDFLAGS) -o $@ $^ -lv4l2 -lv4lconvert -lrt -ljpeg
 
 install: $(TARGETS)
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
