diff -Nuar transmission-2.33.0/libtransmission/rpc-server.c transmission-2.33/libtransmission/rpc-server.c
--- transmission-2.33.0/libtransmission/rpc-server.c	2011-08-21 11:41:51.215297563 +0100
+++ transmission-2.33/libtransmission/rpc-server.c	2011-08-21 11:44:35.248110955 +0100
@@ -46,7 +46,9 @@
  * http://en.wikipedia.org/wiki/Cross-site_request_forgery
  * http://shiflett.org/articles/cross-site-request-forgeries
  * http://www.webappsec.org/lists/websecurity/archive/2008-04/msg00037.html */
-#define REQUIRE_SESSION_ID
+
+/* It looks like lighttpd mod-proxy strips necessary headers -> 409 error */
+#undef REQUIRE_SESSION_ID
 
 #define MY_NAME "RPC Server"
 #define MY_REALM "Transmission"
