diff -r 087c22087187 libparted/arch/linux.c
--- a/libparted/arch/linux.c	Wed Nov 17 02:19:46 2010 +0200
+++ b/libparted/arch/linux.c	Wed Nov 17 02:24:45 2010 +0200
@@ -1701,7 +1701,8 @@
         }
 
         size_t read_length = count * dev->sector_size;
-        if (posix_memalign (&diobuf, dev->sector_size, read_length) != 0)
+        diobuf = malloc(read_length);
+        if (diobuf == NULL)
                 return 0;
 
         while (1) {
@@ -1848,7 +1849,9 @@
 #else
         size_t write_length = count * dev->sector_size;
         dev->dirty = 1;
-        if (posix_memalign(&diobuf, dev->sector_size, write_length) != 0)
+
+        diobuf = malloc(write_length);
+        if (diobuf == NULL)
                 return 0;
         memcpy(diobuf, buffer, write_length);
         diobuf_start = diobuf;
@@ -1905,8 +1908,8 @@
         if (!_device_seek (dev, start))
                 return 0;
 
-        if (posix_memalign(&diobuf, PED_SECTOR_SIZE_DEFAULT,
-                           count * PED_SECTOR_SIZE_DEFAULT) != 0)
+        diobuf = malloc(count * PED_SECTOR_SIZE_DEFAULT);
+        if (diobuf == NULL)
                 return 0;
 
         for (done = 0; done < count; done += status / dev->sector_size) {
